scripts/lib/libvirt/test_libvirt_setup.py

```diff
@@ -120,31 +109,20 @@ class TestLibvirtComputeConfig(unittest.TestCase):
 
     def test_compute_config(self):
         args = arg_parse(
-            ["cloud",
-             "nodecounter",
-             "macaddress",
-             "controller_raid_volumes",
-             "cephvolumenumber",
-             "drbdserial",
-             "computenodememory",
-             "controllernodememory",
-             "libvirttype",
-             "vcpus",
-             "emulator",
-             "vdiskdir"])
-        args.cloud = "cloud"
-        args.nodecounter = "1"
-        args.macaddress = "52:54:01:77:77:01"
-        args.controller_raid_volumes = 0
-        args.cephvolumenumber = "1"
-        args.computenodememory = "2097152"
-        args.controllernodememory = "5242880"
-        args.libvirttype = "kvm"
-        args.vcpus = "1"
-        args.emulator = "/usr/bin/qemu-system-x86_64"
-        args.vdiskdir = "/dev/cloud"
-        args.drbdserial = ""
-        args.bootorder = "3"
+            [("cloud", str, "cloud"),
```

Comment:
@aplanas I tried something like that. The problem is than in this way first we add the arguments, then do the parsing and later we assign the values. ArgParser was complaining on this with int values. When the parser was initiated it dealt with empty strings properly but failed with empty integers (remember: first parsing then assign). Solution for me was to assign the value when starting the parsing. So I think that solution you propose may not work at all.